### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by mikesch803**
1. add auth signup, login
2. add private route wishlist post and delete api

# files to check

- [ ] src/App.js
- [ ] src/context/login-context.jsx
- [ ] src/context/wishlist-context.jsx **(wishlist post and delete api)**
- [ ] src/pages/auth/Auth.js
- [ ] src/pages/auth/NotAuth.js
- [ ] src/pages/login-page/Login.jsx
- [ ] src/pages/signup-page/Signup.jsx
- [ ] src/pages/wishlist-page/Wishlist.jsx **(wishlist get api)**
- [ ] src/reducer/AuthReducer.jsx

### [live link](https://agitated-tesla-b2afa9.netlify.app/)

### CODE REVIEW

File: src/components/product-card/ProductCard.jsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-import { HeartIcon, RatingIcon, WishlistIcon } from "../../icons/icons";
+import { HeartIcon, RatingIcon, WishlistOutlineIcon } from "../../icons/icons";
 import { FilterContext, WishlistContext } from "../../context";
 import { CartContext } from "../../context/cart-context";
 import { Link } from "react-router-dom";
 export function ProductCard() {
   const { getSortedArr } = useContext(FilterContext);
   const { wishlistDataHandler, wishlistData } = useContext(WishlistContext);
   const { dispatch, state } = useContext(CartContext);
+  console.log("wishlistdata ", wishlistData);
```
Comment (ApurvaSawant11) [inline src/components/product-card/ProductCard.jsx]: You can remove this

File: src/context/index.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
-export { FilterContext, FilterProvider } from '../context/filter-context';
-export {WishlistProvider, WishlistContext} from '../context/wishlist-context';
-export { useCart, CartProvider, CartContext } from '../context/cart-context';
+// export { FilterContext, FilterProvider } from './filter-context/filter-context';
+// export {WishlistProvider, WishlistContext} from '../context/wishlist-context';
+// export { useCart, CartProvider, CartContext } from '../context/cart-context';
+// export * from '../context/login-context';
```
Comment (ApurvaSawant11) [inline src/context/index.js]: Remove these comments

File: src/context/wishlist-context.jsx [hunk 3]
```diff
@@ -1,15 +1,56 @@
...
+      try {
+        const response = await axios.delete(
+          `/api/user/wishlist/${product._id}`,
+          {
+            headers: {
+              authorization: encodedToken,
+            },
+          }
+        );
+        console.log("****** ", response);
```
Comment (ApurvaSawant11) [inline src/context/wishlist-context.jsx]: Remove `console.log`'s from this file as well

**Review by ApurvaSawant11 — APPROVED** [review]
There are a lot of `console.log`'s and commented code in all your files. Please check all the files and remove them. Rest everything looks good

File: src/components/product-card/ProductCard.jsx (hunk 1 above)
Comment (mikesch803) [reply src/components/product-card/ProductCard.jsx]: done

File: src/context/wishlist-context.jsx (hunk 3 above)
Comment (mikesch803) [reply src/context/wishlist-context.jsx]: done

File: src/context/index.js (hunk 2 above)
Comment (mikesch803) [reply src/context/index.js]: done

File: src/pages/auth/Auth.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { useContext } from "react"
+import { Navigate, Outlet } from "react-router-dom"
+import { LoginContext } from "../../context"
+
+export const Auth = () => {
+    const {login} = useContext(LoginContext)
+    console.log('from auth ',login)
```
Comment (b0llu) [inline src/pages/auth/Auth.js:7]: remove console log

File: src/pages/auth/NotAuth.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { useContext } from "react"
+import { Navigate, Outlet } from "react-router-dom"
+import { LoginContext } from "../../context"
+
+export const NotAuth = () => {
+    const {login} = useContext(LoginContext)
+    console.log('from auth ',login)
```
Comment (b0llu) [inline src/pages/auth/NotAuth.js:7]: remove console log

File: src/pages/cart-page/Cart.jsx [hunk 6]
```diff
@@ -1,17 +1,29 @@
...
+  //     try {
+  //       const result = await axios.get("/api/user/cart",  {headers: {
+  //         authorization: encodedToken,
+  //       },})
+  //       console.log(result)
+  //     } catch (err) {
+  //       console.log(err);
+  //     }
+  //   })();
+  // }, [encodedToken]);
```
Comment (b0llu) [inline src/pages/cart-page/Cart.jsx:22]: remove commented code if not going to use it

**Review by b0llu — APPROVED** [review]
suggested few things. Everythins else looks good

File: src/pages/cart-page/Cart.jsx (hunk 6 above)
Comment (mikesch803) [reply src/pages/cart-page/Cart.jsx:22]: sure

File: src/pages/auth/NotAuth.js (hunk 5 above)
Comment (mikesch803) [reply src/pages/auth/NotAuth.js:7]: i will do it

File: src/pages/auth/Auth.js (hunk 4 above)
Comment (mikesch803) [reply src/pages/auth/Auth.js:7]: will do it

**Pull request merged**